build/url.go

```diff
@@ -43,7 +44,7 @@ func createTempDockerfileFromURL(ctx context.Context, d *driver.DriverHandle, ur
 		if err != nil {
 			return nil, err
 		}
-		if stat.Size() > 512*1024 {
+		if proto.Size(stat) > 512*1024 {
```

Comment:
This is meant to be file size. We also should probably increase the limit (in a different PR).

Reply:
https://github.com/docker/buildx/pull/2716